### DISCUSSION TITLE
Tab layout for switching between different sections on a Wave dashboard

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#1819] [Q&A] [answered]
Hi!I am working on building a dashboard using Wave. I need to have two sections in the app which can be navigated using tabs. 
1. Is it possible to get this kind of UI?
![image](https://user-images.githubusercontent.com/117831190/215972301-7d19f95c-3286-4526-a1e5-59d9843de5f2.png)

2. How do I create Donut charts in Wave?

**Comment by vopani** [comment, marked as answer]
1. https://wave.h2o.ai/docs/widgets/form/tabs
2. Wave doesn't support donut charts yet. You can open a feature request. As a workaround, you can embed a donut chart made with [matplotlib](https://wave.h2o.ai/docs/examples/plot-matplotlib) or [plotly](https://wave.h2o.ai/docs/examples/plot-plotly) or [bokeh](https://wave.h2o.ai/docs/examples/plot-bokeh)

**Reply by mturoci** [reply]
Wave supports [pie chart](https://wave.h2o.ai/docs/widgets/stat/wide_pie_stat/), but we highly recommend using sorted bar chart for most (if not all) usecases for better UX.

**Reply by n-srinidhi** [reply]
> Wave supports [pie chart](https://wave.h2o.ai/docs/widgets/stat/wide_pie_stat/), but we highly recommend using sorted bar chart for most (if not all) usecases for better UX.

Can you share any example?

**Reply by mturoci** [reply]
https://wave.h2o.ai/docs/widgets/stat/wide_pie_stat/

**Reply by n-srinidhi** [reply]
Thank you! This isn't a sorted bar chart, is it?

**Reply by mturoci** [reply]
Ah, I thought you were asking for the pie char example. A bar chart can be found [here](https://wave.h2o.ai/docs/widgets/plots/interval). By sorted I mean going from the largest to lowest val. Like this

![image](https://user-images.githubusercontent.com/64769322/216009361-6ab1b2a3-3db9-4722-b464-7b896c6214f2.png)

The benefit is that your users do not need to guess which portion is larger if you have for example  (1% and 2% parts) unlike pie/donut.

**Reply by n-srinidhi** [reply]
> 1. https://wave.h2o.ai/docs/widgets/form/tabs
> 2. Wave doesn't support donut charts yet. You can open a feature request. As a workaround, you can embed a donut chart made with [matplotlib](https://wave.h2o.ai/docs/examples/plot-matplotlib) or [plotly](https://wave.h2o.ai/docs/examples/plot-plotly) or [bokeh](https://wave.h2o.ai/docs/examples/plot-bokeh)

How is Form/Tabs different from a tab card? I would like to organize the cards within each Tab section in a specific layout. I am unable to use zones and layout for this if I use Form/Tabs.

**Reply by vopani** [reply]
Tab card is an independent card just for tabs. Often its easier to use tabs within a form card itself. Not much difference in functionality between the two so depends on your UI preference and use case.

Can you describe what you'd like to achieve, maybe with a min code or some screenshots / drawings?
Changing components / cards on a tab switch is pretty straightforward.

**Reply by n-srinidhi** [reply]
Hi, I'm trying to implement something similar to this. A main page or a landing page with content divided into two tabs. On selecting Tab 1,There will be a few charts and tables showing summarized content. On selecting Tab 2, there will be different tables showing different information.
![image](https://user-images.githubusercontent.com/117831190/216256778-812d6ba0-462d-44ff-9a14-360f38dfba77.png)

![image](https://user-images.githubusercontent.com/117831190/216257023-f5ad4b34-3ff6-4677-b75d-194cd1f40aca.png)

**Reply by vopani** [reply]
You can do something like this:

```python
from h2o_wave import main, app, Q, ui

@app('/')
async def serve(q: Q):
    if q.args.tabs == 'tab2':
        del q.page['content_1']
        del q.page['plot_1']
        q.page['content_2'] = ui.form_card(
            box=ui.box(zone='content', order=1),
            items=[
                ui.text(content='Tab 2 table')
            ]
        )
    else:
        q.page['meta'] = ui.meta_card(
            box='',
            layouts=[
                ui.layout(
                    breakpoint='xs',
                    zones=[
                        ui.zone(name='tabs'),
                        ui.zone(name='content', direction='row')
                    ]
                )
            ]
        )

        q.page['tabs'] = ui.form_card(
            box='tabs',
            items=[
                ui.tabs(name='tabs', items=[
                    ui.tab(name='tab1', label='Tab 1'),
                    ui.tab(name='tab2', label='Tab 2')
            ])
        ])

        del q.page['content_2']
        q.page['content_1'] = ui.form_card(
            box=ui.box(zone='content', order=1),
            items=[
                ui.text(content='Tab 1 table')
            ]
        )
        q.page['plot_1'] = ui.form_card(
            box=ui.box(zone='content', order=2),
            items=[
                ui.text(content='Tab 1 plot')
            ]
        )

    await q.page.save()
```

**Reply by n-srinidhi** [reply]
This worked!Thank you!
